machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,154 @@
+	// systemType determines the number of cores and memory that is available.
+	// Few of the supported SystemTypes are s922,e880,e980
+	// e880 systemType available only in Dallas Datacenters
+	// e980 systemType available in Datacenters except Dallas and Washington
+	// Defaults to s922 which is generally available.
+	// +optional
+	SystemType string `json:"systemType,omitempty"`
+
+	// processorType is the VM instance processor type.
+	// It must be set to one of the following values: dedicated, capped or shared.
+	// dedicated: resources are allocated for a specific client, The hypervisor makes a 1:1 binding of a partition’s processor to a physical processor core.
+	// shared: shared among other clients
+	// capped: shared, but resources do not expand beyond those that are requested, the amount of CPU time is capped to the value specified for the entitlement.
+	// if the processorType is selected as dedicated, then processors value cannot be fractional
+	// Defaults to shared.
+	// +kubebuilder:validation:Enum:="dedicated";"shared";"capped"
+	// +optional
+	ProcessorType string `json:"processorType,omitempty"`
+
+	// processors is the number of virtual processors in a virtual machine.
+	// when the processorType is selected as dedicated the processors value cannot be fractional
+	// The minimum processors is 0.5
+	// Defaults to 0.5.
+	// +optional
+	Processors string `json:"processors,omitempty"`
+
+	// memoryGiB is the size of a virtual machine's memory, in GB.
+	// The minimum memory is 32GB
+	// Defaults to 32 GB.
+	// +optional
+	MemoryGiB int32 `json:"memoryGiB,omitempty"`
+}
+
+// PowerVSResource is a reference to a specific PowerVS resource by ID, Name or RegEx
+// Only one of ID, Name or RegEx may be specified. Specifying more than one will result in
+// a validation error.
+// +union
+type PowerVSResource struct {
+	// Type identifies the resource type for this entry.
+	// Valid values are ID, Name and RegEx
+	// +kubebuilder:validation:Enum:=ID;Name;RegEx
+	// +optional
+	Type PowerVSResourceType `json:"type,omitempty"`
+	// ID of resource
+	// +optional
+	ID *string `json:"id,omitempty"`
+	// Name of resource
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// Regex to find resource
+	// Regex contains the pattern to match to find a resource
+	// +optional
+	RegEx *string `json:"regex,omitempty"`
+}
+
+// PowerVSMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It contains PowerVS-specific status information.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PowerVSMachineProviderStatus struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
```

Comment:
Some additional hints
```suggestion
	// +patchMergeKey=type
	// +patchStrategy=merge
	// +listType=map
	// +listMapKey=type
	// +optional
```